### DISCUSSION TITLE
Impossible to get newest version

### DISCUSSION THREAD

**Discussion by kohenkatz** [hjbdev/pvm#24] [General]
Currently, running `pvm install 8.3` installs 8.3.13. However, the newest version on the 8.3.x branch is actually 8.3.14.

The problem is that this program only looks at the downloads listed at https://windows.php.net/downloads/releases/archives/

This list never includes the newest version.

The same is true about 8.4.x - the newest is 8.4.1, but this program can only download 8.4.0.

I think there is a simple fix for this: always check for the requested version at https://windows.php.net/downloads/releases/ before checking https://windows.php.net/downloads/releases/archives/

Here's how this helps:

1. If the user only requested major.minor (no patch), the newest version is always listed at `.../releases/`
2. If the user requested major.minor.patch, it could still be the newest version, but even if it isn't, the only overhead is one additional API call.

A possible additional optimization could be to check https://windows.php.net/downloads/releases/releases.json instead of parsing the directory listing of the `releases` directory. That would mean there would be no need to filter out the "debug", "devel", and "test" downloads. Then check the `archives` directory only for explicitly-requested patch versions that aren't in the JSON file.

I'm going to try to find some time for a PR for this, but I was curious if anyone has any thoughts on it first.

**Comment by hjbdev** [comment]
Seems like a well thought out plan. If you can find time to put up a PR it I'd appreciate it!

**Comment by crabnky** [comment]
I think the correct url is https://downloads.php.net/~windows/releases/archives/ - as I can see it contains all versions. Can you please change it? I want to use pvm, but now I cannot install new versions of php.

**Reply by hjbdev** [reply]
thanks for this! fixed in the latest release

**Reply by crabnky** [reply]
Thanks!

**Comment by kohenkatz** [comment]
@crabnky That's a good link, where did you find it?

At the moment, the official website at https://windows.php.net/download/ is very behind, as is the JSON file that I originally pointed to; the last it has is from December!

**Reply by kohenkatz** [reply]
I'm guessing they stopped updating that site because they now have Windows downloads on the main site at https://www.php.net/downloads.php, but some documentation of that change would have been nice...

**Reply by crabnky** [reply]
I went to the https://www.php.net/downloads.php and examined links for available downloads there. :)
